### ISSUE TITLE
FastQ Screen show all genomes

### ISSUE THREAD

**Issue by AMChalkie** [ewels/MultiQC#829]
**Is your feature request related to a problem? Please describe.**
fastq screen results are only showing a subset of the things searched against.
*  Human
*  Mouse
*  Ecoli_K12
*  Adapters
*  Vectors
*  mouse_rRNA
*  human_rRNA
*  sacCer
*  PhiX
*  dm3
*  seqins_a.r.4

I'm interested in fastq screen results that hit rRNA, and all the libraries I have in my fastq screen database search.

**Describe the solution you'd like**
I would like to see the graph show all the contaminant libraries used in the fastq screen, not just a subset.

**Describe alternatives you've considered**
Allowing selection/removal of specific contaminant libraries

**Comment by ewels** [comment]
Hi @AMChalkie,

I'm pretty sure that the FastQ Screen module doesn't intentionally show a subset. It should already show everything. Please can you send over a set of FastQ Screen results where this is happening?

Thanks,

Phil

**Comment by AMChalkie** [comment]
Thanks for the quick response. On further testing it happens only once the number of samples gets larger (>8 in 2 different directories in my case). 8 in one directory was fine, but past that it goes from vertical bars (where everything is shown), to horizontal bars, where the subset/collapse happens.

**Comment by ewels** [comment]
Ok strange. Are you able to send a bunch of FastQ Screen reports so that I can reproduce this? I’ve never seen this behaviour with the test data I’ve used before. Sounds like a bug in the code somewhere, but I can’t see anything obvious so need to reproduce the behaviour and then poke around to figure out what’s going wrong..

**Comment by ewels** [comment]
Great stuff, thanks! I’ll take a look when I have a chance to sit down with MultiQC. I’m travelling quite a bit in the coming weeks so it could be some time, but I’ll do it ASAP :)

**Comment by ewels** [comment]
Hi @AMChalkie,

I think I figured it out: by default, MultiQC hides empty categories in bar plots to save space / keep things tidy (it makes sense in a surprising number of contexts, honest 😅). What was happening here was that you had lots of hits going into multiple genomes, meaning that a bunch of your references didn't have any _unique_ hits. This put their counts to `0` and they were hidden.

I've changed the config of this plot to always show all categories and now they show up.

Thanks for reporting this! Let me know how you get on with the fix and if you still hit any problems.

Cheers,

Phil
